tests/scripts/analyze_outcomes.py

```diff
@@ -152,9 +176,12 @@ def do_analyze_driver_vs_reference(outcome_file, args):
         'test_function': do_analyze_coverage,
         'args': {}
         },
-    # How to use analyze_driver_vs_reference_xxx locally:
-    # 1. tests/scripts/all.sh --outcome-file "$PWD/out.csv" <component_ref> <component_driver>
-    # 2. tests/scripts/analyze_outcomes.py out.csv analyze_driver_vs_reference_xxx
+    # There are 2 options to use analyze_driver_vs_reference_xxx locally:
+    # 1. Run tests and then analysis:
+    #   - tests/scripts/all.sh --outcome-file "$PWD/out.csv" <component_ref> <component_driver>
+    #   - tests/scripts/analyze_outcomes.py out.csv analyze_driver_vs_reference_xxx
```

Comment:
It's a bit misleading, two same commands will yield different results based on outside, local state. 
Wouldn't it be better to drop `out.csv` in case 2? We don't care what file it uses, as it's internal for the script.

Reply:
Well, I was a little undecided about this as well. The options I had in mind were:

- add a new command line parameter to signal that the python script has to take care of everything (run test + analyze)
- keep the command line as is in order to make this change as transparent as possible

The advantage of keeping the command line "as is" is that you can run multiple tasks without cancelling out the result of the previous one (just pick a different output file's name each time). I don't know how frequent this condition might be, though. Wdyt?

Reply:
I'm not sure if that's useful, since the user can move the output file after each call anyway.
As you've mentioned - since we're adding a new feature, it would be nice to have previous behaviour intact, so I was suggesting an option 3:
- Leave the behaviour as it is right now when there are two arguments;
- If there's only one argument, then the script has to run all the commands (the new feature). 

I'm just afraid that the newly added different behaviour is confusing and might lead to unexpected results if someone, for example, makes a typo in the file name.

That being said, it's not a blocker to me, as I understand that it's rather a matter of opinion and not technical defficiencies, so if you two decide that it's better for you to move on with the current option, I'm fine with it :)

Reply:
Oh, and actually the current version changes the previous behaviour. If someone run the same command twice previously, the out.csv file would be overwritten, whereas now it skips the tests, so the user will have to check each time if `out.csv` is present. It fortunately isn't in the .gitignore file, so at least it's easy to spot :)